runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1449,9 +1449,24 @@ TR_VectorAPIExpansion::isVectorizedOrScalarizedNode(TR::Node *node, TR::DataType
    int32_t refId = -1;
 
    if (node->getOpCodeValue() == TR::aload ||
-       node->getOpCodeValue() == TR::astore)
-      {
-      refId = node->getSymbolReference()->getReferenceNumber();
+       node->getOpCodeValue() == TR::astore ||
+       (node->getOpCode().isVectorOpCode() &&
+        (node->getOpCode().getVectorOperation() == TR::vload ||
+         node->getOpCode().getVectorOperation() == TR::vstore)
+        )) {
+
+      if (node->getOpCodeValue() == TR::aload ||
+          node->getOpCodeValue() == TR::astore) {
+          refId = node->getSymbolReference()->getReferenceNumber();
+      }
+      else {
+          TR::SymbolReference *origSymRef = _nodeTable[node->getGlobalIndex()]._origSymRef;
+
+          if (!origSymRef)
+              return false;  // was vectorized earlier by auto-SIMD or another pass of VectorAPIExpansion
+
+          refId = origSymRef->getReferenceNumber();
```

Comment:
This logic for `vload` and `vstore` was already going to happen in the `isVectorOpCode()` case below. I suppose the point of this change is that the resulting `refId` should be subject to all of the same `_aliasTable`-related checks as we use for `aload` and `astore`? Could you please elaborate on what's going on in a comment and/or the commit message?

Reply:
Yes, the code below will only handle methods now. You are right, we should handle vload/vstore the same way as aload/astore except for the way we get the symbol reference. I will add to the message.